niwidgets/streamlines.py

```diff
@@ -0,0 +1,208 @@
+from __future__ import print_function
+import nibabel as nib
+import numpy as np
+from ipywidgets import interact, fixed, widgets
+import os
+import ipyvolume as ipv
+
+
+def length(x):
+    """Returns the sum of euclidean distances between neighboring points"""
+    return np.sum(np.sqrt(np.sum((x[:-1, :] - x[1:, :]) *
+                                 (x[:-1, :] - x[1:, :]),
+                                 axis=1)))
+
+
+def color(x):
+    """Returns an approximation for color of line based on its endpoints"""
+    dirvec = (x[0, :] - x[-1, :])
+    return (dirvec/(np.sqrt(np.sum(dirvec*dirvec, axis=-1)))).dot(np.eye(3))
+
+
+class StreamlineWidget:
+    """
+    Turns mrtrix track or trackvis files into interactive plots using ipyvolume.
```

Comment:
The safest here would be to say that this can use any file type that nibabel knows how to read. This will be true even for future versions that implement reading of any other streamline file.

Reply:
roger.